pkg/cmd/cluster/list.go

```diff
@@ -0,0 +1,96 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"github.com/spf13/cobra"
+	"github.com/zeet-dev/cli/pkg/api"
+	"github.com/zeet-dev/cli/pkg/cmdutil"
+	"github.com/zeet-dev/cli/pkg/iostreams"
+	"gopkg.in/yaml.v2"
+)
+
+type ClusterListOptions struct {
+	IO           *iostreams.IOStreams
+	ApiClient    func() (*api.Client, error)
+	OutputFormat string
+
+	Team string
+}
+
+func NewClusterListCmd(f *cmdutil.Factory) *cobra.Command {
+	var opts = &ClusterListOptions{}
+	opts.IO = f.IOStreams
+	opts.ApiClient = f.ApiClient
+
+	cmd := &cobra.Command{
+		Use: "list [team]",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.Team = args[0]
+			}
+
+			return runClusterList(opts)
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&opts.OutputFormat, "output", "o", "text", "format: text|json|yaml|table")
+
+	return cmd
+}
+
+func runClusterList(opts *ClusterListOptions) error {
+	client, err := opts.ApiClient()
+	if err != nil {
+		return err
+	}
+
+	clusters, err := client.ListClusters(context.Background(), opts.Team)
+	if err != nil {
+		return err
+	}
+
+	switch opts.OutputFormat {
+	case "text":
+		for _, cluster := range clusters {
+			fmt.Fprintln(opts.IO.Out, cluster.ID, cluster.Name)
+		}
+	case "json":
+		output, err := json.MarshalIndent(clusters, "", "  ")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
```

Comment:
Doesn't returning the error out to the cobra command cause it to print to Stdout? I think this will cause the error to print to both stdout and stderr. If we want to print the error only to stderr we should print here and return nil

Reply:
that is a interesting question, since we are expecting user to run this thru other utils like pipe and jq. what is the standard way to handle error happens somewhere in the upstream util like zeet cli? do we return exit code 1 or empty json list or print error message to stdout/stderr. is there some standard i can reference. to me it feels like all these options are equally valid.

Reply:
i think we should always send errors to stderr; this way there's no chance of a user accidentally piping error messages into downstream programs. I think this is not currently implemented basically anywhere in our CLI (certainly not for errors that are simply returned out to the Cobra command, but that is hopefully a very small change)

beyond that, i've seen wildly divergent behavior. but i think a reasonable rule for us is that for any given step (either a full command or a step in some interactive workflow within a command) we should either return data to stdout, OR an error to stderr plus a non-zero exit code, but not both.

For this particular PR I propose that we skip the printing to stderr, simply `return err` to allow cobra to handle this. Then we should make a ticket to improve error handling and make it consistent